mantidproject/mantid
# Reflectometry GUI Restructuring: feature branch merge

## Description
This PR is for the final merge of the long-running feature branch `isis_reflectometry_alternative_table` into master. All code in this feature branch has been reviewed in individual PRs already. The main tasks for reviewing this PR will therefore be final sanity checking of the overall changes, checking that documentation etc. is up to date, and thorough testing.

Any further fixes to be included in this PR should not be made on this branch directly but be made on separate branches and merged into `isis_reflectometry_alternative_table` via their own PRs.

## Summary of changes
The `qt/scientific_interfaces/ISISReflectometry` directory has been reorganised into the following subdirectories:
- `Common`: some generic supporting functions
- `Reduction`: this centres around the reduction model, which acts as a configuration for the reduction. It is the job of the presenters to build and maintain this representation.
- `GUI`: The GUI components have been reorganised as follows:
  - The [DataProcessorWidget](https://docs.mantidproject.org/nightly/interfaces/DataProcessorWidget.html) in the `RunsTableView` has been replaced with the [JobTreeView](https://developer.mantidproject.org/BatchWidget/index.html).
  - The Settings tab has been split into separate Experiment and Instrument tabs.
  - The tabs are grouped into one or more "Batches", rather than having multiple "groups" of settings within each tab. The new structure is shown below.

![image](https://user-images.githubusercontent.com/12895056/60197905-4e413e80-9838-11e9-8b87-6153c17740ab.png)

## To test

- Testing should be done by someone at ISIS.
- The [unscripted testing](https://www.mantidproject.org/Testing_ISIS_Reflectometry_GUI) for the GUI would be a good place to start.
- Testing should be done in both MantidPlot and Workbench.
- We should check that the SANS GUI still works as expected - particularly the runs table.
- We should test any other areas that might be affected by changed code (I think primarily this is anything that uses `BatchAlgorithmRunner` and possibly `BoostOptionalToAlgorithmProperty.h`)

Refs #23051 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
